mlrun/feature_store/steps.py

```diff
@@ -52,6 +54,9 @@ def __init__(self, **kwargs):
     def do(self, event):
         """
         This method defines the do method of this class according to the first event type.
+
+        .. warning::
+            When extending this class, do not override this method - only override the `_do_XXX` methods.
```

Comment:
When extending this class, do not override this method; only override the `_do_XXX` methods.